### ISSUE TITLE
Control which tests may run in parallel with which other tests

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit#165]
Implement Exclusion groups of tests such that only one test in a group may run at a time.

**Comment by jnm2** [comment]
I proposed a potential design before realizing this was the right place to do it. https://github.com/nunit/nunit/issues/2389

**Comment by MaceWindu** [comment]
Would be nice to have it.
In our project we have tests, where each test executed against multiple databases.
Only one test could be run against each database at the same time. 
Being able to to schedule tests betwen test runners based on some discriminator whould speed up our test run a lot.

**Comment by Darkproduct** [comment]
Same for me. I write full feature tests that run different test benches depending on the hardware that is needed. I could just separate all tests into different fixtures for each test bench, but there are a lot of tests, that have the same test code but a different setup.
For me, it would be perfect if I could give each fixture a runner scope and all tests with different scopes can run in parallel and all tests with the same scope have to run sequentially.

**Comment by maettu-this** [comment]
I favor a string based approach, something like `[HasParallelScope("Foo")]` and `[IncompatibleParallelScope("Foo")]` as proposed in #2389, ideally even supporting wild-carded `[IncompatibleParallelScope("Foo*")]`, but I'd prefer extending existing NUnit attributes like e.g.:

`Parallelizable(ParallelScope.Children, "Foo")]` and `[NonParallelizable("Foo*")]`

Where `[NonParallelizable("*")]` has the same result as today's `[NonParallelizable]`.
